jax/_src/lax/lax.py

```diff
@@ -3681,7 +3686,13 @@ def _float_to_int_for_sort(x):
   signed_dtype = _INT_DTYPES[nbits]
   unsigned_dtype = _UINT_DTYPES[nbits]
 
+  # Standardize NaNs: NaN is the only float value that does not equal itself.
+  x = select(ne(x, x), _nans(x), x)
```

Comment:
I don't know if we have `lax.isnan(x)` but it would probably make sense to define it (like this) if we don't.

Reply:
Good idea - I made it a private utility for now. We can expose it in the `lax` namespace in a followup.